sql/core/src/test/scala/org/apache/spark/sql/connector/AlterTableTests.scala

```diff
@@ -310,6 +311,26 @@ trait AlterTableTests extends SharedSparkSession {
     }
   }
 
+  test("SPARK-39383 DEFAULT columns on V2 data sources with ALTER TABLE ADD COLUMN") {
+    withSQLConf(SQLConf.DEFAULT_COLUMN_ALLOWED_PROVIDERS.key -> s"$v2Format, ") {
```

Comment:
Hmm, why setting the "v2Format" is not enough?

Reply:
Good Q. The tests use the `InMemoryTableCatalog` which passes an empty table provider for intermediate in-memory tables. By allowing both the `v2Format` as well as the empty string, both tables built `s"using $v2Format"` and in-memory catalog tables used in testing are supported.

Reply:
Got it:)